### PR TITLE
Alerting: Align notifier truncation and logging with prometheus/alertmanager

### PR DESCRIPTION
**Pull request by alexweav**
Some external systems have limits on message sizes. Prometheus handles this in a unified way. We recently touched this area in upstream.

The way we do truncation and log is very inconsistent in Grafana notifiers, even among each other.

**Why do we need this feature?**

This PR aligns Grafana's notifier field truncation with upstream. We now use the same logic in the same places, same constants, and emit the same logs.

Effectively, ports https://github.com/prometheus/alertmanager/pull/3135 and https://github.com/prometheus/alertmanager/pull/3145 to Grafana, plus some additional alignment that was missing entirely.

**Who is this feature for?**

Contributes to alignment of alertmanagers.

**Which issue(s) does this PR fix?**:

Contributes to https://github.com/grafana/alerting-squad/issues/304
Contributes to https://github.com/grafana/alerting-squad/issues/240

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/util.go [hunk 1]
```diff
@@ -317,3 +317,53 @@ func splitCommaDelimitedString(str string) []string {
...
+
+	if n <= 3 {
+		return string(r[:n]), true
+	}
+
+	return string(r[:n-1]) + truncationMarker, true
+}
+
+// TruncateInBytes truncates a string to fit the given size in Bytes.
+// TODO: This is more advanced than the upstream's TruncateInBytes. We should consider upstreaming this, and removing it from here.
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/util.go:341]: Copied verbatim from `webex.go`. I added the TODO so we remember to do this in the future.

File: pkg/services/ngalert/notifier/channels/pushover.go [hunk 2]
```diff
@@ -206,6 +219,27 @@ func (pn *PushoverNotifier) genPushoverBody(ctx context.Context, as ...*types.Al
...
+	message := tmpl(pn.settings.message)
+	message, truncated = TruncateInRunes(message, pushoverMaxMessageLenRunes)
+	if truncated {
+		pn.log.Warn("Truncated message", "incident", key, "runes", pushoverMaxMessageLenRunes)
+	}
+	message = strings.TrimSpace(message)
+	if message == "" {
+		// Pushover rejects empty messages.
+		message = "(no details)"
+	}
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/pushover.go:235]: taken from Prometheus

**Comment by alexweav** [comment]
lint error resolved in https://github.com/grafana/grafana/pull/59432

**Review by yuri-tceretian — COMMENTED** [review]
LGTM

**Review by yuri-tceretian — APPROVED** [review]

**Comment by alexweav** [comment]
Upstream PR is now merged, this is unblocked. Few minor changes to uptake.

**Pull request merged**